### PR TITLE
CI: checksums cleanup

### LINKED ISSUE #5206: CTest follow-up: add missing checksum files
Follow-up on #5068. Add missing checksum files that were uncovered in the migration to CTest.

### PR DESCRIPTION
**Pull request by EZoni**
Fix #5206 and prepare for #5256:
- [x] Add missing checksums
  - [x] `test_1d_background_mcc_picmi`
  - [x] `test_2d_background_mcc_picmi`
  - [x] `test_3d_particle_absorption`
  - [x] `test_2d_dirichlet_bc`
  - [x] `test_2d_dirichlet_bc_picmi`  
  - [x] `test_2d_field_probe` 
  - [x] `test_3d_embedded_boundary_picmi`
  - [x] `test_2d_particle_reflection_picmi` 
  - [x] `test_rz_laser_acceleration_opmd` 
  - [x] `test_2d_runtime_components_picmi` 
- [x] Clean up calls to `evaluate_checksum`

Calls to `evaluate_checksum` are now uniform across the analysis scripts, making it easier to automatize further changes. Hopefully the uniformity will be preserved by copy/pasting the necessary code:
- ```py
  import os
  import sys
  ```
- ```py
  sys.path.insert(1, "../../../../warpx/Regression/Checksum/")
  from checksumAPI import evaluate_checksum
  ```
- ```py
  # compare checksums
  evaluate_checksum(
      test_name=os.path.split(os.getcwd())[1],
      output_file=sys.argv[1],
  )
  ```
  with this last one possibly modified by adding other arguments, e.g., `output_format`, `rtol`, `do_particles`, etc.

### CODE REVIEW

File: Examples/Physics_applications/capacitive_discharge/analysis_1d.py [hunk 1]
```diff
@@ -2,8 +2,14 @@
 
 # Copyright 2022 Modern Electron, David Grote
 
+import os
+import sys
+
 import numpy as np
 
+sys.path.append("../../../../warpx/Regression/Checksum/")
```
Comment (EZoni) [inline Examples/Physics_applications/capacitive_discharge/analysis_1d.py]:
```suggestion
sys.path.insert(1, "../../../../warpx/Regression/Checksum/")
```

File: Examples/Physics_applications/capacitive_discharge/analysis_2d.py [hunk 2]
```diff
@@ -7,12 +7,15 @@
 # solver that directly solves the Poisson equation using matrix inversion
 # rather than the iterative approach from the MLMG solver.
 
+import os
 import sys
 
 sys.path.append("../../../../warpx/Regression/Checksum/")
```
Comment (EZoni) [inline Examples/Physics_applications/capacitive_discharge/analysis_2d.py]:
```suggestion
sys.path.insert(1, "../../../../warpx/Regression/Checksum/")
```

File: Examples/Physics_applications/capacitive_discharge/analysis_dsmc.py [hunk 3]
```diff
@@ -8,14 +8,7 @@
 import numpy as np
 
 sys.path.append("../../../../warpx/Regression/Checksum/")
```
Comment (EZoni) [inline Examples/Physics_applications/capacitive_discharge/analysis_dsmc.py]:
```suggestion
sys.path.insert(1, "../../../../warpx/Regression/Checksum/")
```

File: Examples/Physics_applications/laser_acceleration/analysis_1d_fluid_boosted.py [hunk 4]
```diff
@@ -24,8 +24,8 @@
 import numpy as np
 from scipy.constants import c, e, epsilon_0, m_e
 
-sys.path.insert(1, "../../../../warpx/Regression/Checksum/")
-import checksumAPI
+sys.path.append("../../../../warpx/Regression/Checksum/")
```
Comment (EZoni) [inline Examples/Physics_applications/laser_acceleration/analysis_1d_fluid_boosted.py]:
```suggestion
sys.path.insert(1, "../../../../warpx/Regression/Checksum/")
```

**Review by ax3l — APPROVED** [review]

File: Regression/Checksum/checksum.py [hunk 5]
```diff
@@ -243,9 +243,8 @@ def evaluate(self, rtol=1.0e-9, atol=1.0e-40):
             )
             print("Benchmark: %s" % ref_benchmark.data.keys())
             print("Test file: %s" % self.data.keys())
-            print("\n----------------\nNew file for " + self.test_name + ":")
+            print(f"\nNew checksums file {self.test_name}.json:")
             print(json.dumps(self.data, indent=2))
-            print("----------------")
             sys.exit(1)
```
Comment (ax3l) [inline Regression/Checksum/checksum.py:249]: Please double check if this script by @NeilZaim still works when this output here is changed:
https://warpx.readthedocs.io/en/latest/developers/checksum.html#reset-a-benchmark-from-the-azure-pipeline-output-on-github

Comment (EZoni) [reply Regression/Checksum/checksum.py:249]: Thanks for pointing this out. I'm working on a fix in https://github.com/ECP-WarpX/WarpX/pull/5372.

**Pull request merged**